scripts/post_migration_fixes/pmf_report_type_assignments_20220308_in1172/annual_report_type_assignments_update.sql

```diff
@@ -0,0 +1,83 @@
+CREATE SCHEMA IF NOT EXISTS pmf_report_type_assignments_20220308_in1172;
+
+-- Populate table with data we're going to change
+SELECT *
+INTO pmf_report_type_assignments_20220308_in1172.annual_report_type_assignments_updates
+FROM (
+
+    SELECT
+        annual_report_type_assignments_id,
+        annualreport_id,
+        reporttype AS original_reporttype,
+        (CASE
+            WHEN
+                orderstatus = 'ACTIVE' AND supervisionlevel = 'GENERAL'
+            THEN
+                CASE
+                    WHEN assetlevel IN ('HIGH', 'UNKNOWN') THEN 'OPG102'
+                    WHEN assetlevel = 'LOW' THEN 'OPG103'
+                    ELSE NULL
+                END
+            ELSE
+                NULL
+        END) AS expected_reporttype,
+        type AS original_type,
+        'pfa' AS expected_type
+    FROM (
+        SELECT
+            arta.id AS annual_report_type_assignments_id,
+            arl.id AS annualreport_id,
+            arta.reporttype,
+            arta.type,
+            c.id AS order_id,
+            c.orderstatus AS orderstatus,
+            row_number() OVER (
+                PARTITION BY p.caserecnumber
+                ORDER BY arl.reportingperiodenddate DESC
+            ) AS rownum,
+            sll.supervisionlevel AS supervisionlevel,
+            sll.assetlevel AS assetlevel
+        FROM annual_report_logs arl
+        INNER JOIN persons p
+        ON arl.client_id = p.id
+        INNER JOIN cases c
+        ON p.caserecnumber = c.caserecnumber
+        INNER JOIN annual_report_type_assignments arta
+        ON arl.id = arta.annualreport_id
+        LEFT JOIN supervision_level_log sll
+        ON c.id = sll.order_id
+        WHERE arl.status = 'PENDING'
+        AND c.type = 'order'
+        AND p.clientsource IN ('CASRECMIGRATION')
+    ) AS reports
+    WHERE rownum = 1
+
+) to_update;
+
+-- Populate audit table
+SELECT arta.*
+INTO pmf_report_type_assignments_20220308_in1172.annual_report_type_assignments_updates_audit
+FROM pmf_report_type_assignments_20220308_in1172.annual_report_type_assignments_updates up
+INNER JOIN annual_report_type_assignments arta
+ON up.annual_report_type_assignments_id = arta.id;
+
+BEGIN;
+    UPDATE annual_report_type_assignments
```

Comment:
Hmm, good point again. Yes, probably, as these will be new entries in annual_report_logs which otherwise won't have an annual_report_type_assignments row.

Reply:
Then again... All I need to do is update the existing reports with a type assignment, then look for PENDING reports which don't have a type assignment (and are associated with the migration). @devkokov are you adding any new annual_report_logs or just removing incorrect ones?

Reply:
Just removing incorrect ones, so yeah you will need to update and insert, or you could delete all and insert.